vms/platformvm/warp/validator_test.go

```diff
@@ -305,41 +304,3 @@ func TestSumWeight(t *testing.T) {
 		})
 	}
 }
-
-func BenchmarkGetCanonicalValidatorSet(b *testing.B) {
```

Comment:
Could we keep this test with fetching the validator set itself mocked?

I think we should have benchmarks for:
- `GetCanonicalValidatorSet` with `GetValidatorSet` mocked as it is here
- `GetValidatorSet` isolated with different depths and different amounts of total activity across those blocks (we should get an idea of the expected and worst case performance)
- (optional) benchmark for the combination of the two

Separate from the benchmarks, we should also gather metrics and profile on testnet/mainnet to see the performance in practice.